### DISCUSSION TITLE
Searchlight with Crossnobis

### DISCUSSION THREAD

**Discussion by jasminstein** [rsagroup/rsatoolbox#330] [Q&A]
Hi everyone,

I am trying to run a searchlight analyses. What I want is to compute searchlight RDMs across multiple runs using cross-validated square Mahalanobis distance. However, the get_searchlight_RDMs function takes in a 2d data input. What does my input data need to look like for this function to compute the crossnobis in a searchlight? Is that even possible?

Cheers,
Jasmin

**Comment by JonathanShor** [comment]
Just a bump for this. Is there any theoretical challenge to using the full crossnobis method with searchlight, or its just not been implemented (yet)?

**Reply by JasperVanDenBosch** [reply]
Not implemented yet. A larger re-work of the searchlight (and other "mass" pipelines) I something that I've been wanting to work on for a while now.

**Reply by JonathanShor** [reply]
I'm going to expand to `get_searchlight_RDMs` to accept `cv_descriptor` and `noise`, including selecting the correct pieces of `noise` for each center & neighbors. I'll likely use something like #329 as well for, but with the two-for-one `process_map` from `tqdm.contrib.concurrent` instead of `Parallel` directly.

I figure this goes in `searchlight.py`, and I see that most of the work done in #253 so far is adjacent. I'll share once its done and you can let me know if you'd like a PR or something.

**Reply by JonathanShor** [reply]
@JasperVanDenBosch I have a working draft in this branch: https://github.com/JonathanShor/rsatoolbox/tree/enhance_searchlight

It works for full crossnobis, including CV and basic noise handling, or any other current metric `calc_rdm` accepts.
I have not developed any tests for it, and it may not totally adhere to any format conventions you use, tho it is documented.

How can I most usefully contribute this? Should I open a new PR for it?
